photon-core/src/main/java/org/photonvision/common/configuration/NeuralNetworkProperties.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) Photon Vision.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.photonvision.common.configuration;
+
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Optional;
+import org.photonvision.common.configuration.NeuralNetworkModelManager.Family;
+import org.photonvision.rknn.RknnJNI;
+
+public class NeuralNetworkProperties {
```

Comment:
Consider renaming this to better match the fact that this thing maintains global state in `HashMap<Path, ModelProperties> properties`